### PR TITLE
build: update dependencies for v1.1.0 release

### LINKED ISSUE #97: Update project dependencies for `v1.1.0` release
Code editors like Visual Studio Code provide excellent built-in Node application debugging, and this dependency triggers warnings during the install process due to regressions, therefore, as a pre-release task for `v1.1.0`, the `debug` dependency should be removed.

In addition to removing this dependency, other out-of-date or vulnerable dependencies may also be addressed as they apply.

### PR DESCRIPTION
**Pull request by BlueSlug**
* [x] This pull request has been linted by running `npm run lint` without errors
* [x] This pull request has been tested by running `npm run start` and reviewing affected routes
* [x] This pull request has been built by running `npm run build` without errors
* [x] This isn't a duplicate of an existing pull request

## Description

This PR updates direct and indirect dependencies to address security vulnerabilities. In addition to updating, the `debug` dependency has been removed altogether.

## Steps to test

1. run `npm ci` to install the dependencies as specified by `package-lock.json`
2. run `npm run build`
3. run `npm run lint`
4. run `npm start`

**Expected behavior:** The install, build and dev server launch processes should run without issue. Once launched, the site should function as before.

## Additional information

In my environment, the `commitlint` package seems not to be functioning currently, so my commit messages are not being checked for compliance with Conventional Commits. After some testing, I confirmed that this is the case from not only the `build/update-deps-GH-97` branch, but also `dev` and `main`.

This issue has been resolved in #85, so one of the branches will need to be updated after the other is merged into `dev` (depending on the order).

Unrelated to this issue, two direct dependencies were left alone in order to avoid getting mired in ES Modules and other significant changes just before release. These deps are `move-file` and `husky`. Husky has been updated in #85, and a separate issue will be filed to address `move-file` and ESM conversion at a later date.

## Related issues

Resolves #97

### CODE REVIEW

**Comment by cindyli** [comment]
The pull request looks good. The small issue is the `linkedom` module had a new upgrade to `0.14.8` after you finished this work. In this PR, `linkedom` is using a lower version of `0.14.7`. Can you upgrade it? Thanks.

**Comment by BlueSlug** [comment]
Ready for more review :)

**Comment by jobara** [comment]
@BlueSlug how far are you from the 1.1.0 release? I'm wondering if this should be the last thing merged so that if other dependencies update prior to the release they can be captured as well.

File: package.json [hunk 1]
```diff
@@ -30,28 +30,26 @@
         }
     },
     "dependencies": {
-        "@11ty/eleventy": "1.0.0",
+        "@11ty/eleventy": "1.0.1",
         "@11ty/eleventy-navigation": "0.3.2",
-        "@11ty/eleventy-plugin-syntaxhighlight": "3.2.2",
+        "@11ty/eleventy-plugin-syntaxhighlight": "4.0.0",
         "eleventy-plugin-fluid": "1.0.0",
-        "infusion": "4.0.0",
```
Comment (jobara) [inline package.json:37]: Are you sourcing Infusion from `eleventy-plugin-fluid`?

Comment (BlueSlug) [reply package.json:37]: Infusion isn't directly referenced by anything in the website itself any more (now that the call to `fluid.allocateSimpleId()` was removed from #85), though I was able to find the following versions being pulled in as subdeps in the `package-lock`:

| package | version |
|----------|---------|
| eleventy-plugin-fluid | `4.x` |
| fluid-glob | `3.0.0-dev.20210312T233557Z.0b016a6dc.FLUID-6580` |
| fluid-lint-all | `4.0.0` |
| markdownlint-config-fluid | `3.0.0-dev.20210312T233557Z.0b016a6dc.FLUID-6580` |

So the short answer seems to be yes, but it also seems to be sourced from several other packages as well.

**Comment by BlueSlug** [comment]
@cindyli ready for review. As discussed earlier today, I've incorporated the latest round of Dependabot updates but will not do another round of direct dependency updates for this release

**Pull request merged**